app/components/projects/settings/life_cycle_steps/step_component.html.erb

```diff
@@ -3,16 +3,14 @@
     align_items: :center,
     justify_content: :space_between
   ) do |step_container|
-    step_container.with_column(flex_layout: true, mr: 2, classes: "min-width-0") do |title_container|
-      title_container.with_column(pt: 1, mr: 1, classes: icon_color_class) do
-        render Primer::Beta::Octicon.new(icon: icon)
-      end
-      title_container.with_column(pt: 1, mr: 2, classes: "ellipsis") do
-        render(Primer::Beta::Text.new(classes: "filter-target-visible-text")) { definition.name }
-      end
-      title_container.with_column(pt: 1, classes: "no-wrap") do
-        render(Primer::Beta::Text.new(**gate_text_options)) { gate_info }
-      end
+    step_container.with_column(mr: 2, classes: "min-width-0") do
+      render(
+        Projects::PhaseDefinitionComponent.new(
+          definition,
+          edit_link: User.current.admin?,
+          phase_text_options: { classes: "filter-target-visible-text" }
```

Comment:
I like how the `"filter-target-visible-text"` is logically separated to be context aware, but I'm unsure if it is required, because we seem to use in all the instances. Do you think we will use this component without the `"filter-target-visible-text"` in the future?

Reply:
@dombesz I don't know. I just felt that it is not the responsibility of that component to be ready to be filterable with that controller.